app/grandchallenge/components/tasks.py

```diff
@@ -1139,3 +1139,106 @@ def validate_voxel_values(*, civ_pk):
             civ.image.save()
 
     civ.interface._validate_voxel_values(civ.image)
+
+
+@shared_task(
+    **settings.CELERY_TASK_DECORATOR_KWARGS["acks-late-micro-short"],
+)
+def add_image_to_object(
+    *,
+    app_label,
+    model_name,
+    object_pk,
+    interface_pk,
+    upload_session_pk,
+):
+    from grandchallenge.components.models import (
+        ComponentInterface,
+        ComponentInterfaceValue,
+    )
+
+    with transaction.atomic():
+        object = get_model_instance(
+            pk=object_pk,
+            app_label=app_label,
+            model_name=model_name,
+        )
+        interface = ComponentInterface.objects.get(pk=interface_pk)
+        upload_session = RawImageUploadSession.objects.get(
+            pk=upload_session_pk
+        )
+        try:
+            image = Image.objects.get(origin_id=upload_session_pk)
+        except (Image.DoesNotExist, Image.MultipleObjectsReturned):
+            error_message = "Image imports should result in a single image"
+            upload_session.status = RawImageUploadSession.FAILURE
+            upload_session.error_message = error_message
+            upload_session.save()
+            return
+
+        object.values.remove(*object.values.filter(interface=interface))
+        civ, created = ComponentInterfaceValue.objects.get_or_create(
+            interface=interface, image=image
+        )
+        if created:
```

Comment:
Add some breathing room:

```suggestion

        if created:
```